draft-ietf-rats-corim.md

```diff
@@ -1429,9 +1429,173 @@ is extended using Endorsements etc. from the accepted tags.
+the entries.
+
+This specification defines the matching algorithm for some CBOR tagged reference
+values, which is described in sub-sections below.
+
+A CoRIM profile may define additional tags and their matching algorithms.
+
+If the Verifier does not recognize the Reference Value CBOR tag value then
+the Reference Value does not match.
+
+If the Reference Value is not tagged and the measurement-value-map key is a
+value with handling described in the sub-sections below,
+then the algorithm appropriate to that key is used to match the entries.
+
+If the Reference Value is not tagged, and the `measurement-values-map` key
+is not a value described below, then the entries are compared
+using binary comparison of their CBOR encoded values. If the values
+are not binary identical then the Reference Value does not match.
+
+Note that while specifications may extend the matching semantics using CBOR tags,
+there is no way to extend the matching semantics of key values.
+Any new keys requiring non-default comparison must add a CBOR tag to the
+reference value describing the desired behaviour.
+
+If the Reference Value contains an `authorized-by` field then the Verifier
+SHALL check for an `authorized-by` field in the Accepted Claims Set entry
+compared in the steps above. If the Accepted Claims Set key is not one of
+the keys from the reference value `authorized-by` field then the
+reference value does not match.
+
+If all checks above have been performed successfully then the Reference Value
+matches.
+
+##### Comparison for svn entries
+
+The value stored under `measurement-values-map` key 1 is an SVN, which must
+have type UINT.
+
+If the Reference value for `measurement-values-map` key 1 is an untagged UINT or
+a UINT tagged with #6.552 then an equality comparison is performed. If the value
+of the SVN in Accepted Claims Set is not equal to the value in the Reference
+Value then the Reference Value does not match.
+
+If the Reference value for `measurement-values-map` key 1 is a UINT tagged with
+#6.553 then a minimum comparison is performed. If the value of the SVN in
+Accepted Claims Set less than the value in the Reference Value then the
+Reference Value does not match.
+
+##### Comparison for digests entries
+
+The value stored under `measurement-values-map` key 2, or a value tagged with
+#6.TBD is a digest entry. It contains one or more digests, each measuring the
+same object. A Reference Value may contain multiple digests, each with a
+different algorithm acceptable to the Reference Value provider. If the
+digest in evidence contains a single value with an algorithm and value
+matching one of the algorithms and values in the Reference Value then it
+matches.
+
+To prevent downgrade attacks, if there are multiple algorithms which are in
+both the evidence and Reference Value then the digests calculated using all
+shared algorithms must match.
+
+If the CBOR structure of the digest entry in the Reference Value or the
```

Comment:
```suggestion
If the CBOR structure of the `digests` entry in the Reference Value or the
```

Reply:
Couldn't an implementation convert the CBOR array of `digests` into a linked list or some other memory structure and loose the CBOR encoding (so long as it preserves array position and byte order)?